packages/nextjs/src/client/performance.ts

```diff
@@ -83,13 +85,23 @@ function extractNextDataTagInformation(): NextDataTagInfo {
   nextDataTagInfo.params = query;
 
   if (props && props.pageProps) {
-    if (props.pageProps._sentryBaggage) {
-      nextDataTagInfo.baggage = props.pageProps._sentryBaggage;
+    const sentryTrace = props.pageProps._sentryTraceData;
+    const baggage = props.pageProps._sentryBaggage;
+
+    const { traceparentData, dynamicSamplingContext, propagationContext } = tracingContextFromHeaders(
+      sentryTrace,
+      baggage,
+    );
+
+    if (dynamicSamplingContext) {
+      nextDataTagInfo.dynamicSamplingContext = dynamicSamplingContext;
     }
 
-    if (props.pageProps._sentryTraceData) {
-      nextDataTagInfo.traceParentData = extractTraceparentData(props.pageProps._sentryTraceData);
+    if (traceparentData) {
+      nextDataTagInfo.traceParentData = traceparentData;
     }
+
+    getCurrentHub().getScope().setPropagationContext(propagationContext);
```

Comment:
m: Since this function is currently pure i would move this into nextRouterInstrumentation directly.